### PR TITLE
Fix message editing.

### PR DESCRIPTION
**Pull request by vishwesh3**
- Fix: edit message not working.  |   |   | 4cc5abd
- Show a dialog if edit message time limit is over. |   |   | 9e27c77
- replaceEmoticonsWithEmoji in the edited message content. |   |   | 4de31ca
- Do nothing if edited content and original content is same. |   |   | 77a6efd
- Focus input after selecting Edit message. |   |   |  edf7b7b

screen shot 2017-08-14 at 11 30 12 am

### CODE REVIEW

File: src/compose/ComposeBox.js [hunk 1]
```diff
@@ -108,6 +110,21 @@ export default class ComposeBox extends PureComponent {
     this.clearMessageInput();
   };
 
+  handleEdit = () => {
+    const { auth, editMessage, actions } = this.props;
+    const { message } = this.state;
+    if (editMessage.content !== message) {
+      try {
+        patchMessage(auth, replaceEmoticonsWithEmoji(message), editMessage.id, (msg: string) =>
```
Comment (borisyankov) [inline src/compose/ComposeBox.js]: Why did you add a callback and not use the try catch block?

Comment (vishwesh3) [inline src/compose/ComposeBox.js]: It was not working, so I thought of adding callback function.
Now refactored it a bit and is now working 👍 .

**Comment by borisyankov** [comment]
Good job. Thanks.
